tests/validators/test_bytes.py

```diff
@@ -0,0 +1,27 @@
+import re
+
+import pytest
+
+from pydantic_core import SchemaValidator, ValidationError
+
+from ..conftest import Err
+
```

Comment:
you probably want to start with a general parametrised test which shows which types work and which do not.

You will also need to add a test (parametrised or not) which uses `py_or_json`) and therefore tests bytes fields from JSON.